### PR TITLE
🌱 hack/e2e don't add binary files to artifacts and also censor base64 encoded values

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Improves our censoring in e2e.sh to:

* not add binary files (e.g. gzipped files) to the output artifacts because we can't control their data
* also censor the base64 version of our secret

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: hack/e2e.sh [hunk 1]
```diff
@@ -50,11 +50,17 @@ on_exit() {
...
       done || true
+      VSPHERE_PASSWORD_B64=$(echo -n "${VSPHERE_PASSWORD}" | base64 -b0)
+      grep -I -r -l -e "${VSPHERE_PASSWORD_B64}" "${ARTIFACTS}" | while IFS= read -r file
+      do
+        echo "Cleaning up VSPHERE_PASSWORD_B64 from file ${file}"
+        sed -i "s/${VSPHERE_PASSWORD_B64}/REDACTED/g" "${file}"
+      done || true
     fi
     # Move all artifacts to the original artifacts location.
     mv "${ARTIFACTS}"/* "${ORIGINAL_ARTIFACTS}/"
```
Comment (sbueringer) [inline hack/e2e.sh:66]: Doesn't this still mv all files?

Comment (chrischdi) [reply hack/e2e.sh:66]: ah, you're totally right 🤔

File: hack/e2e.sh [hunk 2]
```diff
@@ -49,12 +49,27 @@ on_exit() {
 
   # Cleanup VSPHERE_PASSWORD from temporary artifacts directory.
   if [[ "${ORIGINAL_ARTIFACTS}" != "" ]]; then
+    # Delete non-text files from artifacts directory to not leak files accidentially
```
Comment (sbueringer) [inline hack/e2e.sh:52]: Can we make a bit surer that this is only run over ARTIFACTS? :) Just because it's rm.. :D

Comment (chrischdi) [reply hack/e2e.sh:52]: You're too quick 😂

**Comment by sbueringer** [comment]
/test pull-cluster-api-provider-vsphere-e2e-govmomi-main

(just want to run it with more data)

**Comment by sbueringer** [comment]
/lgtm
/approve

/hold
@chrischdi We don't even run "Cleaning up VSPHERE_PASSWORD" anymore?

**Comment by sbueringer** [comment]
I would guess linux vs mac base64 binary

**Comment by chrischdi** [comment]
>     failed to get component source YAML from URL: failed to get https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.6.4/core-components.yaml: got status code 502

/retest

**Comment by sbueringer** [comment]
Thx. Really appreciate the work on this!

/lgtm
/approve

**Comment by chrischdi** [comment]
Tests are failing due to the on-going capi release.

See: https://github.com/kubernetes-sigs/cluster-api/issues/10450

**Pull request merged**